### PR TITLE
[Merged by Bors] - feat: port Data.Set.List + leanprover-community/mathlib#18647

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/List/Lemmas.lean [hunk 1]
```diff
@@ -23,30 +23,6 @@ variable {α β γ : Type _}
...
-#align list.range_map_coe List.range_map_coe
-
-/-- If each element of a list can be lifted to some type, then the whole list can be
-lifted to this type. -/
-instance canLift (c) (p) [CanLift α β c p] :
-    CanLift (List α) (List β) (List.map c) fun l => ∀ x ∈ l, p x where
-  prf l H := by
-    rw [← Set.mem_range, range_map]
-    exact fun a ha => CanLift.prf a (H a ha)
-#align list.can_lift List.canLift
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/Data/List/Lemmas.lean:48]: Nevermind, I'm just blind

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks, this all looks fine to me.

@urkud, if you see something I missed, please make a follow up. Since the absence of the new file is muddling the porting dashboard I think it's best to just merge it now.

**Comment by eric-wieser** [comment]
bors d+

**Comment by Parcly-Taxel** [comment]
bors merge

**Pull request closed without merging**